src/qt/addresstablemodel.cpp

```diff
@@ -563,21 +563,27 @@ int AddressTableModel::lookupAddress(const QString& address) const
  * @return
  */
 QString AddressTableModel::getLastUnusedAddress() const{
+    QString addressStr;
     LOCK(wallet->cs_wallet);
     if(!wallet->mapAddressBook.empty()) {
-        for (std::map<CTxDestination, AddressBook::CAddressBookData>::iterator it = wallet->mapAddressBook.end(); it != wallet->mapAddressBook.begin(); --it) {
-            if(it != wallet->mapAddressBook.end()) {
-                if (it->second.purpose == "receive") {
+        for (auto it = wallet->mapAddressBook.rbegin(); it != wallet->mapAddressBook.rend(); ++it ) {
+            if(it != wallet->mapAddressBook.rend()) {
```

Comment:
this is not needed (you already have `it != wallet->mapAddressBook.rend()` in the `for` condition).
Anyways, as stated in the comment, you can just iterate normally from the beginning.